test_data/config.toml

```diff
@@ -482,3 +482,143 @@ post_cmd = '''
     cp .flox/env/manifest.toml "$env_dir/manifest.toml"
     cp .flox/env/manifest.lock "$env_dir/manifest.lock"
 '''
+
+# lock (and only lock) hello-unfree
+# used to test buildenv's ability to build unfree packages
+[envs.hello-unfree-lock]
+cmd = "true # no-op"
+post_cmd = '''
+echo '
+version = 1
+install.hello-unfree.pkg-path = "hello-unfree"
+' | flox lock-manifest - > "$RESPONSE_FILE"
```

Comment:
(non-blocking) It's confusing to repurpose `$RESPONSE_FILE` to be a flat JSON representation of `manifest.lock` when it's normally an array of catalog HTTP responses. Could you write a follow-up ticket to move these to a "locks" section in `mk_data`?